src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Random getRandom() {
+        return random;
+    }
+
+    public Figure getRandomFigure() {
+
+        String figureColor = colorSupplier.getRandomColor();
+        int figuresCount = random.nextInt(FIGURE_COUNT);
+
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        int bottomSideNumber = random.nextInt(FIGURE_COUNT);
+        int topSideNumber = random.nextInt(FIGURE_COUNT);
+        int leftSideNumber = random.nextInt(FIGURE_COUNT);
+        int rightSideNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 1:
+                Figure square = new Square("Square", figureColor,
+                        bottomSideNumber, bottomSideNumber,
+                        bottomSideNumber, bottomSideNumber);
```

Comment:
lets generate different random numbers for different figure parameters. Same for all figures